program/rust/src/time_machine_types.rs

```diff
@@ -7,10 +39,91 @@ pub struct TimeMachineWrapper {
     place_holder: [u8; 1864],
 }
 
+impl Tracker for TimeMachineWrapper {
+    fn add_first_price(
+        &mut self,
+        _current_time: u64,
+        _current_price: i64,
+        _current_conf: u64,
+    ) -> Result<(), OracleError> {
+        msg!("implement me");
+        Ok(())
+    }
+    fn add_price(
+        &mut self,
+        _current_time: u64,
+        _prev_time: u64,
+        _current_price: i64,
+        _prev_price: i64,
+        _current_conf: u64,
+        _prev_conf: u64,
+    ) -> Result<(), OracleError> {
+        msg!("implement me");
+        Ok(())
+    }
+}
+
+#[derive(Copy, Clone)]
+#[repr(C)]
+/// wraps everything stored in a price account
+pub struct PriceAccountWrapper {
+    //an instance of the c price_t type
+    pub price_data:            pc_price_t,
+    //space for more publishers
+    pub extra_publisher_space: [u8; EXTRA_PUBLISHER_SPACE as usize],
+    //TimeMachine
+    pub time_machine:          TimeMachineWrapper,
+}
+impl PriceAccountWrapper {
+    #[cfg(feature = "resize-account")]
+    pub fn add_first_price(&mut self) -> Result<(), OracleError> {
```

Comment:
I think this function and the one below should verify that the current status == TRADING

Reply:
Right, I do not need to check the previous status but it is always the last with trading status, but for the current one, there's no such gurantee, thanks

Reply:
I made so that add_price verifies that things have been initialized in addition to that